### PR TITLE
Copy framkework

### PR DESCRIPTION
**Pull request by lmsilvera**
#287

---
This change is [Reviewable](https://reviewable.io/reviews/opensalt/opensalt/322)

### CODE REVIEW

**Comment by roverwolf** [comment]
Reviewed 12 of 12 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[bin/encore, line 14 at r1](https://reviewable.io:443/reviews/opensalt/opensalt/322#-L4D6qzrrFqe1nLn3btS:-L4D6qzrrFqe1nLn3btT:bpjtpz0) ([raw file](https://github.com/opensalt/opensalt/blob/5ee47bb94e384ba368d0e24d85fcf8ce1a129634/bin/encore#L14)):*
>     ./app/Resources/assets/js/lsdoc/index.js \

Why was `lsdoc/index.js` moved from the main webpack config to being merged with these files?

The objective is to reduce the number of files being merged together here.

---

*[src/CftfBundle/Controller/CopyController.php, line 26 at r1](https://reviewable.io:443/reviews/opensalt/opensalt/322#-L4D0AYN69TlJIeBIxiP:-L4D0AYN69TlJIeBIxiQ:b-qmr1lq) ([raw file](https://github.com/opensalt/opensalt/blob/5ee47bb94e384ba368d0e24d85fcf8ce1a129634/src/CftfBundle/Controller/CopyController.php#L26)):*
> ```PHP
>      * @Route("/framework/{id}", name="copy_framework_content")
>      * @Method("POST")
>      *
> ```

`CopyController:frameworkAction()` should have an `@Security` annotation (or checked inside the method) to restrict it to the users allowed to make the copy.

---

*[src/CftfBundle/Repository/LsDocRepository.php, line 348 at r1](https://reviewable.io:443/reviews/opensalt/opensalt/322#-L4DAknfjRNHj38nKvTD:-L4DAknfjRNHj38nKvTE:b-xxicbm) ([raw file](https://github.com/opensalt/opensalt/blob/5ee47bb94e384ba368d0e24d85fcf8ce1a129634/src/CftfBundle/Repository/LsDocRepository.php#L348)):*
> ```PHP
>         }
> 
>         echo count($oldLsDoc->getAssociationGroupings());
> ```

Is the `echo` statement left over from debugging?

---

*[src/CftfBundle/Resources/views/DocTree/view.html.twig, line 417 at r1](https://reviewable.io:443/reviews/opensalt/opensalt/322#-L4D1dFqAn_bTod0MIce:-L4D1dFqAn_bTod0MIcf:bl018sx) ([raw file](https://github.com/opensalt/opensalt/blob/5ee47bb94e384ba368d0e24d85fcf8ce1a129634/src/CftfBundle/Resources/views/DocTree/view.html.twig#L417)):*
> <div class="modal fade" id="copyFrameworkModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
> 			<div class="modal-dialog modal-lg" role="document">
> 				<div class="modal-content">

Tabs here should be spaces.

---

*[tests/acceptance/features/framework/super_user/copy_framework.feature, line 5 at r1](https://reviewable.io:443/reviews/opensalt/opensalt/322#-L4D1k0WRqkyq9p96XJQ:-L4D1k0Xi_0z51DPgOcp:b-x8jn5m) ([raw file](https://github.com/opensalt/opensalt/blob/5ee47bb94e384ba368d0e24d85fcf8ce1a129634/tests/acceptance/features/framework/super_user/copy_framework.feature#L5)):*
> ```Gherkin
>   @super-user @framework @copy-framework @1016-1323 @duplicate
>   Scenario: An Super-User can copy a framework
>     Given I log in as a user with role "Super-User"
> ```

If only one test is being added, it should be for the minimum-privileged user that can do the action.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/322)*

**Comment by roverwolf** [comment]
Reviewed 11 of 12 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[tests/acceptance/features/framework/organization_editor/copy_framework.feature, line 4 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/322#-L4Ips2pXItDornJfpwi:-L4Ips2pXItDornJfpwj:bicptos) ([raw file](https://github.com/opensalt/opensalt/blob/07f5992ed8045d31afc3ea89cbd2f246593e9cf2/tests/acceptance/features/framework/organization_editor/copy_framework.feature#L4)):*
> Scenario: 1013-1408 An organization editor can copy a framework

The id of the scenario in each of the tests should be unique to the specific test (and added as @scenario-id as a tag).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/322)*

File: app/Resources/assets/js/lsdoc/index.js [hunk 1]
```diff
@@ -631,6 +631,8 @@ var Util = (function () {
     };
 })();
 
+apx.Util = Util;
```
Comment (roverwolf) [inline app/Resources/assets/js/lsdoc/index.js]: Setting `apx.Util` fails on pages that do not have `apx` defined (such as the front page).

**Comment by roverwolf** [comment]
Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/322)*

**Comment by roverwolf** [comment]
One thing that would be nice would be to clean up (delete) the newly created frameworks at the end of the tests.

---

Reviewed 3 of 3 files at r4, 6 of 6 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[tests/acceptance/features/framework/organization_admin/copy_framework.feature, line 11 at r5](https://beta.reviewable.io/reviews/opensalt/opensalt/322#-L4NjpLyBYrnkODuL8Xb:-L4NjpLyBYrnkODuL8Xc:b-u6hwsb) ([raw file](https://github.com/opensalt/opensalt/blob/6a6abc4dea5c41bc06df7c5a0d9fc86d8f2c4150/tests/acceptance/features/framework/organization_admin/copy_framework.feature#L11)):*
>     When I click on copy framework
>     Then I should see a success message

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/opensalt/opensalt/322#-:-L4Njtz_7UPpVSYTH7UR:b-6xpm6r)*

**Comment by roverwolf** [comment]
@lmsilvera Please update the branch and resolve the merge conflict.

Other than that, is this PR ready to merge?

**Comment by lmsilvera** [comment]
@roverwolf PR request updated and ready to be merged

**Pull request merged**